src/db/migrations/migration2.js

```diff
@@ -0,0 +1,120 @@
+const pLimit = require('p-limit');
+const async = require('async');
+const logger = require('../../utils/logger');
+const DBHelper = require('../db-helper');
+const { TX_STATUS, EVENT_STATUS } = require('../../constants');
+const MultipleResultsEventApi = require('../../api/multiple-results-event');
+const EventLeaderboard = require('../../models/event-leaderboard');
+const GlobalLeaderboard = require('../../models/global-leaderboard');
+
+const PROMISE_CONCURRENCY_LIMIT = 15;
+const limit = pLimit(PROMISE_CONCURRENCY_LIMIT);
+
+async function wrapper(dbMethod, callback) {
+  await dbMethod();
+  callback();
+}
+
+async function migration2(next) {
+  try {
+    // migrate event leaderboard
+    const events = await DBHelper.findEvent({ txStatus: TX_STATUS.SUCCESS });
+    const promises = [];
+
+    events.forEach((event, key) => {
+      promises.push(limit(async () => {
+        const eventAddress = event.address;
+        const resultSet = await DBHelper.findResultSet({ eventRound: 0, txStatus: TX_STATUS.SUCCESS, eventAddress: event.address });
+        const bets = await DBHelper.findBet({
+          eventAddress: event.address,
+          txStatus: TX_STATUS.SUCCESS,
+        });
+        const txs = resultSet.concat(bets);
+
+        try {
+          let i = 0;
+          await async.whilst(
+            check => check(null, i < txs.length), // trigger iter
+            (next) => {
```

Comment:
I just tested, you can use `async () => {`. See here for how to implement: https://caolan.github.io/async/v3/global.html

```
await async.whilst(
            check => check(null, i < txs.length), // trigger iter
            async () => {
              const tx = txs[i];
              i++;
              const userAddress = tx.betterAddress || tx.centralizedOracleAddress;
              const eventLeaderboardEntry = new EventLeaderboard({
                eventAddress,
                userAddress,
                investments: tx.amount,
                winnings: '0',
              });
              await DBHelper.insertEventLeaderboard(eventLeaderboardEntry);
              if (event.status === EVENT_STATUS.WITHDRAWING) {
                const globalLeaderboardEntry = new GlobalLeaderboard({
                  userAddress,
                  investments: tx.amount,
                  winnings: '0',
                });
                await DBHelper.insertGlobalLeaderboard(globalLeaderboardEntry);
              }
            },
          );
```